### PR TITLE
Add getSupportedItemTypesOfChannel method to StateProfileType

### PR DESCRIPTION
**Pull request by triller-telekom**
This method is necessary to query whether a StateProfileType is applicable
to a certain ChannelType with its supportedItemTypes.

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

**Comment by triller-telekom** [comment]
close/reopen due to unrelated test failure

**Comment by triller-telekom** [comment]
@kaikreuzer I have addressed your comments.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/SystemProfiles.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
     StateProfileType FOLLOW_TYPE = ProfileTypeBuilder.newState(FOLLOW, "Follow").build();
 
     StateProfileType OFFSET_TYPE = ProfileTypeBuilder.newState(OFFSET, "Offset").withSupportedItemTypes("Number")
-            .build();
+            .withSupportedItemTypesOfChannel("Number").build();
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/SystemProfiles.java]: Btw, would be nice if you could use the constant `CoreItemFactory.NUMBER`.´(also in the call of withSupportedItemTypes).

Comment (triller-telekom) [reply bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/SystemProfiles.java]: Ah we have constants for the item types, nice. Thanks for pointing this out to me. I have added a commit that addresses it.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/StateProfileTypeImpl.java [hunk 2]
```diff
@@ -52,4 +55,9 @@ public String getLabel() {
         return label;
     }
 
+    @Override
+    public Collection<String> getSupportedItemTypesOfChannel() {
+        return supportedItemTypesOfChannel;
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/StateProfileTypeImpl.java]: The caller is able to modify the collection.
I don't think this is the desired behavior. Is it?
Why not holding a unmodifiable view only, so no need to act here but once in the constructor only (perhaps copy the provided collection once depend on the builder impl.).

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/TriggerProfileTypeImpl.java [hunk 3]
```diff
@@ -41,4 +46,19 @@ public TriggerProfileTypeImpl(ProfileTypeUID profileTypeUID, String label, Colle
         return supportedChannelTypeUIDs;
     }
 
+    @Override
+    public Collection<String> getSupportedItemTypes() {
+        return supportedItemTypes;
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/TriggerProfileTypeImpl.java]: See above

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/StateProfileType.java [hunk 4]
```diff
@@ -12,15 +12,25 @@
...
  */
 @NonNullByDefault
 public interface StateProfileType extends ProfileType {
 
+    /**
+     * Get a collection of ItemType names that a Channel needs to support in order to able to use this ProfileType
+     *
+     * @return a collection of supported ItemType names (an empty list means ALL types are supported)
+     */
+    Collection<String> getSupportedItemTypesOfChannel();
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/StateProfileType.java]: Shouldn't that method be used at any place or do we introduce it for future use only?

Comment (maggu2810) [reply bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/StateProfileType.java]: It is used by #6025

**Pull request merged**